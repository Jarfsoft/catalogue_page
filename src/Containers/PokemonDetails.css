.poke-details {
  background-color: #282c34;
  padding: 2rem 5rem;
}

.poke-details-card {
  background-color: white;
  margin: 2rem auto;
  border: solid 1px black;
  border-radius: 15px;
  padding: 2rem;
  width: 90%;
  height: 27rem;
  display: flex;
}

.profile {
  text-align: center;
  width: auto;
  padding: 2rem;
  margin: auto 1rem;
  border: solid 1px black;
  border-radius: 15px;
}

.item1 { grid-area: id; }
.item2 { grid-area: type; }
.item3 { grid-area: height; }
.item4 { grid-area: weight; }
.item5 { grid-area: abilities; }

.details-data {
  text-align: center;
  display: grid;
  grid-template-areas:
    'id type'
    'height weight'
    'abilities abilities'
    'abilities abilities';
  grid-template-rows: 4rem 4rem 8rem;
  width: 100%;
  margin: 3rem 2rem;
}

.details-data p {
  height: 1rem;
}

.details-data b {
  margin: 0 1rem;
}

.abilities {
  width: 75%;
  margin: 0 auto;
}

.abilities ul {
  display: grid;
  grid-template-columns: repeat(4, 7rem);
}

.abilities li {
  width: 6rem;
  margin: 1rem;
}
