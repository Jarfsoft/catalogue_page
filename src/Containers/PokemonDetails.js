import { React, useEffect, useState } from 'react';
import { Link, useParams } from 'react-router-dom';

import { getPokemon } from '../pokeApi';
import './PokemonDetails.css';

function PokemonDetails() {
  const params = useParams();

  const capitalize = (s) => {
    if (typeof s !== 'string') return '';
    return s.charAt(0).toUpperCase() + s.slice(1);
  };

  const [data, setData] = useState();

  useEffect(() => {
    getPokemon(params.id, setData);
  }, []);

  return (
    <div className="poke-details">
      <div className="poke-details-card">
        <div className="profile">
          <h1>{data ? capitalize(data.name) : ''}</h1>
          <img src={data ? data.sprites.front_default : ''} alt="Pokemon sprite" />
        </div>
        <div className="details-data">
          <p className="item1">
            <b>ID:</b>
            {data ? data.id : ''}
          </p>
          <p className="item2">
            <b>Type:</b>
            {data ? capitalize(data.types[0].type.name) : ''}
          </p>
          <p className="item3">
            <b>Height:</b>
            {data ? `${data.height} in` : ''}
          </p>
          <p className="item4">
            <b>Weight:</b>
            {data ? `${data.weight} lb` : ''}
          </p>
          <div className="item5">
            <b>Abilities:</b>
            <br />
            <div className="abilities">
              <ul>
                {data ? data.moves.slice(0, 10).map((p) => <li key={p.move.name}>{p.move.name}</li>) : ''}
              </ul>
            </div>
          </div>
        </div>
        <Link to="/">Back</Link>
      </div>
    </div>
  );
}

export default PokemonDetails;
