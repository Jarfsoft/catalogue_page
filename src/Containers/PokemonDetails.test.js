import React from 'react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { BrowserRouter } from 'react-router-dom';
import PokemonDetails from './PokemonDetails';

const initialState = {
  list: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12],
  type: '',
};

jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useSelector: jest.fn()
    .mockReturnValueOnce(initialState),
}));

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useParams: () => ({
    id: 1,
  }),
}));

describe('PokemonDetails container', () => {
  const mockStore = configureStore();
  let store;

  it('PokemonDetails page is rendered', () => {
    store = mockStore(initialState);
    const pokemondetails = renderer
      .create(
        <Provider store={store}>
          <BrowserRouter>
            <PokemonDetails />
          </BrowserRouter>
        </Provider>,
      ).toJSON();
    expect(pokemondetails).toMatchSnapshot();
  });

  it('should contain a image', () => {
    const pokemondetails = render(
      <Provider store={store}>
        <BrowserRouter>
          <PokemonDetails />
        </BrowserRouter>
      </Provider>,
    );
    const image = pokemondetails.container.querySelectorAll('img');
    expect(image).toHaveLength(1);
  });
});
