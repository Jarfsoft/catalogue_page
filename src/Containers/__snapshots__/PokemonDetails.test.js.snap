// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonDetails container PokemonDetails page is rendered 1`] = `
<div
  className="poke-details"
>
  <div
    className="poke-details-card"
  >
    <div
      className="profile"
    >
      <h1>
        
      </h1>
      <img
        alt="Pokemon sprite"
        src=""
      />
    </div>
    <div
      className="details-data"
    >
      <p
        className="item1"
      >
        <b>
          ID:
        </b>
        
      </p>
      <p
        className="item2"
      >
        <b>
          Type:
        </b>
        
      </p>
      <p
        className="item3"
      >
        <b>
          Height:
        </b>
        
      </p>
      <p
        className="item4"
      >
        <b>
          Weight:
        </b>
        
      </p>
      <div
        className="item5"
      >
        <b>
          Abilities:
        </b>
        <br />
        <div
          className="abilities"
        >
          <ul>
            
          </ul>
        </div>
      </div>
    </div>
    <a
      href="/"
      onClick={[Function]}
    >
      Back
    </a>
  </div>
</div>
`;
